cmd/booster-http/server.go

```diff
@@ -164,6 +169,16 @@ func (s *HttpServer) handleByPieceCid(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// TODO we need to get a payment "key" from the request, which we can use to query our go-nitro node with. It's just stubbed out here for now.
+	paymentInfo := s.nitroClient.GetVirtualChannel(types.Destination{})
+
+	if paymentInfo.Balance.PaidSoFar.ToInt().Cmp(big.NewInt(0)) < 0 { // TODO we need to compare against some "going rate" or agreed-upon price for the retrieval
```

Comment:
```suggestion
	if paymentInfo.Balance.PaidSoFar.ToInt().Cmp(RETRIEVAL_PRICE_PER_BYTE*NUM_BYTES) < 0 { // TODO we need to compare against some "going rate" or agreed-upon price for the retrieval
```

Reply:
We're going to need some more advanced book keeping than this. We wouldn't want a single payment (or channel) to be able to be reused for multiple retrievals (i.e. double spending problem).

One idea is to have an object that we can query with a unique retrieval id, to see if it has been paid for or not. That object can wrap the nitro rpc client and abstract away all of the details.